### PR TITLE
Add flock and lockf to fcntl module

### PR DESCRIPTION
**Pull request by fanninpm**

### CODE REVIEW

File: stdlib/src/fcntl.rs [hunk 1]
```diff
@@ -2,6 +2,8 @@ pub(crate) use fcntl::make_module;
 
 #[pymodule]
 mod fcntl {
+    use std::mem;
+
```
Comment (youknowone) [inline stdlib/src/fcntl.rs:6]:
```suggestion
```

Comment (fanninpm) [reply stdlib/src/fcntl.rs]: Accepted with revisions in 2c8588d.

File: stdlib/src/fcntl.rs [hunk 2]
```diff
@@ -8,6 +8,7 @@ mod fcntl {
         stdlib::{io, os},
         PyResult, VirtualMachine,
     };
+    use std::mem;
```
Comment (youknowone) [inline stdlib/src/fcntl.rs:11]: unused import warning here. To avoid cfg, I left another suggestion

File: stdlib/src/fcntl.rs [hunk 3]
```diff
@@ -139,4 +140,74 @@ mod fcntl {
...
+    #[pyfunction]
+    fn lockf(
+        fd: i32,
+        cmd: i32,
+        len: OptionalArg<PyIntRef>,
+        start: OptionalArg<PyIntRef>,
+        whence: OptionalArg<i32>,
+        vm: &VirtualMachine,
+    ) -> PyResult {
+        let mut l: libc::flock = unsafe { mem::zeroed() };
```
Comment (youknowone) [inline stdlib/src/fcntl.rs:167]:
```suggestion
        let mut l: libc::flock = unsafe { std::mem::zeroed() };
```

**Review by youknowone — APPROVED** [review]

**Pull request merged**